ui/espresso/BasicSample/app/src/main/java/com/example/android/testing/espresso/BasicSample/MainActivity.java

```diff
@@ -54,16 +54,17 @@ public void onClick(View view) {
         // Get the text from the EditText view.
         final String text = mEditText.getText().toString();
 
-        switch (view.getId()) {
-            case R.id.changeTextBt:
-                // First button's interaction: set a text in a text view.
-                mTextView.setText(text);
-                break;
-            case R.id.activityChangeTextBtn:
-                // Second button's interaction: start an activity and send a message to it.
-                Intent intent = ShowTextActivity.newStartIntent(this, text);
-                startActivity(intent);
-                break;
+        final int changeTextBtId = R.id.changeTextBt;
```

Comment:
Unrelated change?

Reply:
This is required. Bazel doesn't finalize R ids at the `android_library` level in order for `aapt` to process/merge the resources at the `android_binary` level. [This is a good explanation](https://blog.bazel.build/2018/02/14/how-android-builds-work-in-bazel.html) of why Bazel needs this:

> The resource ID values generated for android_library targets are only temporary, since higher-level targets might depend on multiple targets where different resources were assigned the same ID. To ensure that resource IDs aren’t persisted anywhere permanent, the R files record the IDs as nonfinal, ensuring that compilation doesn’t inline them into other Java code. Additionally, an android_library's R files should be discarded after building is complete.